lxd/container_lxc.go

```diff
@@ -1898,6 +1918,89 @@ func (c *containerLXC) initLXC(config bool) error {
 	return nil
 }
 
+func (c *containerLXC) deviceStart(deviceName string, config map[string]string, instanceRunning bool) (device.RunConfig, error) {
+	d, err := device.New(c, config)
+	if err != nil {
+		return device.RunConfig{}, err
+	}
+
+	if instanceRunning && !d.CanHotPlug() {
+		return device.RunConfig{}, fmt.Errorf("Device cannot be started when instance is running")
+	}
+
+	runConfig, err := d.Start(c.deviceVolatileGet(deviceName), func(save map[string]string) error {
+		return c.deviceVolatileSet(deviceName, save)
+	})
+
+	if err != nil {
+		return device.RunConfig{}, err
+	}
+
+	return runConfig, nil
+}
+
+func (c *containerLXC) deviceUpdate(deviceName string, config map[string]string, oldConfg map[string]string, instanceRunning bool) error {
```

Comment:
@stgraber device live update function, which has been wired into the newly created updateNetworkDevice.